src/mqttcm_connect.c

```diff
@@ -1619,7 +1805,7 @@ int AddToSubscriptionList(char *compName, char *topic)
 		comp_topic_name_t* temp = g_head;
 		while (temp != NULL)
 		{
-			if(strcmp(compName, temp->compName))
+			if(strcmp(compName, temp->compName) == 0)
```

Comment:
AddToSubscriptionList() is called only when Subscribe is needed, why we do we need isSubscribeNeeded() check again inside the func ?
also why we set temp->subscribeOnFlag to 1 here, in newNode creation it is already enabled while subscribing at line 1824 .

Reply:
I removed the subscribe flag set from all other places and added inside subcribeId function so only when successful subscribe happens it will add the flag one

Reply:
ok , subscribe flag looks correct. but isSubscribeNeeded() does nt look proper

Reply:
In optimization phase we can remove this check under AddToSubscriptionList and to use in mqtt_subscribe alone

Reply:
ok